### PR TITLE
Jeremiah

### PR DESCRIPTION
**Pull request by wannaBeKakashi**
Remaining with tests.

### CODE REVIEW

File: app/Http/Controllers/Api/CourseController.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+class CourseController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index() :AnonymousResourceCollection
+    {
+        //
+        $course = Course::all();
```
Comment (enocknkhoma) [inline app/Http/Controllers/Api/CourseController.php:23]: Rename $course to $courses  on line # 23

File: app/Http/Controllers/Api/CourseController.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(CreateRequest $request) : CourseResource
+    {
+        //
+        $create_course = $request->post();
```
Comment (enocknkhoma) [inline app/Http/Controllers/Api/CourseController.php:41]: rename create $create_course to $data line #41

File: app/Http/Controllers/Api/CourseController.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        $course = Course::create($create_course);
+        return new CourseResource($course);
+    }
+
+    /**
+     * Display the specified resource.
+     */
+    public function show(string $id)
```
Comment (enocknkhoma) [inline app/Http/Controllers/Api/CourseController.php:49]: The show method should return the Resource wit that ID

File: app/Http/Controllers/Api/CourseController.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        //
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     */
+    public function edit(string $id)
```
Comment (enocknkhoma) [inline app/Http/Controllers/Api/CourseController.php:57]: Remove the show and edit method

File: app/Http/Controllers/Api/CourseController.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        //
+    }
+
+    /**
+     * Update the specified resource in storage.
+     */
+    public function update(UpdateRequest $request, string $id)
```
Comment (enocknkhoma) [inline app/Http/Controllers/Api/CourseController.php:65]: an update should return the resource 
rename $update_course to $updates

File: app/Http/Requests/CourseComment/CreateRequest.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\Rule|array|string>
+     */
+    public function rules(): array
+    {
+        return [
```
Comment (enocknkhoma) [inline app/Http/Requests/CourseComment/CreateRequest.php:24]: user id and course id are also required

File: routes/api.php [hunk 7]
```diff
@@ -17,3 +22,19 @@
 Route::middleware('auth:sanctum')->get('/user', function (Request $request) {
     return $request->user();
 });
```
Comment (enocknkhoma) [inline routes/api.php:26]: Use middleware group to group all routes that requires sanctum

File: app/Http/Resources/ContentCommentResource.php [hunk 8]
```diff
@@ -0,0 +1,19 @@
+<?php
+
```
Comment (enocknkhoma) [inline app/Http/Resources/ContentCommentResource.php:2]: Put the resources in their own name space 
i.e CourseResource goes to Resources/Course/CourseResource

**Review by enocknkhoma — CHANGES_REQUESTED** [review]
@wannaBeKakashi , please look at the attached comments

File: tests/Feature/ContentCommentTest.php [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+use Illuminate\Http\Response;
+use Tests\TestCase;
+
+class ContentCommentTest extends TestCase
+{
+    use RefreshDatabase;
+    /**
+     * A basic feature test example.
+     */
+    public function test_can_list_all_content_comments(): void
```
Comment (enocknkhoma) [inline tests/Feature/ContentCommentTest.php:17]: Chief, how are you authenticating the user making the request to the APIs in the test. ? 
Look at how this was implemented in the ES_MMS

File: tests/Feature/CourseContentTest.php [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+        $response = $this->getJson('api/course_contents');
+
+        $response->assertStatus(200);
+
+        $this->assertCount(10, $response->decodeResponseJson()['data']);
+    }
+    public function test_can_create_course_content(): void
+    {
+        $course_content = CourseContent::factory()->make()->toArray();
```
Comment (enocknkhoma) [inline tests/Feature/CourseContentTest.php:29]: rename $course_content to $data

File: tests/Feature/CourseContentTest.php [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+        $response = $this->postJson('api/course_contents',$course_content);
+
+        $response->assertStatus(Response::HTTP_CREATED);
+    }
+    public function test_can_update_course_content(): void 
+    {
+        $course_content = CourseContent::factory()->create();
```
Comment (enocknkhoma) [inline tests/Feature/CourseContentTest.php:37]: rename course_update to $updates

**Review by enocknkhoma — CHANGES_REQUESTED** [review]

**Comment by enocknkhoma** [comment]
@wannaBeKakashi , so let's close the above issues and then we close on this one

File: routes/api.php (hunk 7 above)
Comment (wannaBeKakashi) [reply routes/api.php:26]: An error comes which i include sanctum

Comment (wannaBeKakashi) [reply routes/api.php:26]: when*

**Review by enocknkhoma — APPROVED** [review]

**Pull request merged**